common/djangoapps/oauth_exchange/views.py

```diff
@@ -35,3 +43,26 @@ def post(self, request, _backend):
             edx_access_token = self.create_access_token(request, user, scope, client)
 
         return self.access_token_response(edx_access_token)
+
+
+class SessionCookieExchangeView(APIView):
+    """
+    View for exchanging an access token for session cookies
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    @method_decorator(csrf_exempt)
+    def get(self, request):
+        request.user.backend = get_backend_for_user(request.user)
+        login(request, request.user)  # login generates and stores the user's cookies in the session
+        return HttpResponse(status=204)  # cookies stored in the session are returned with the response
+
+
+def get_backend_for_user(user):
```

Comment:
I think a better name for this would be get_arbitrary_backend_for_user, and it needs a docstring explaining its purpose. I had to read through the django code to figure out what the backend was used for to make sure this was harmless.